packages/strapi-plugin-upload/admin/src/containers/HomePage/reducer.js

```diff
@@ -22,13 +22,13 @@ const reducer = (state, action) => {
       return state.removeIn(['dataToDelete', index]);
     }
     case 'TOGGLE_SELECT_ALL': {
-      const { value } = action;
+      const isSelected = List(state.get('data')).size === List(state.get('dataToDelete')).size;
```

Comment:
Here the type should be a `List` so you can do state.get('data').size directly. To do so you will need to change the case `GET_DATA_SUCCCEEDED` by:

```
state.update('data', () => fromJS(action.data)):
```

Reply:
Agree !

Reply:
You create your state with the fromJS function. So `data` should be an immutable List 
![Capture d’écran 2020-03-13 à 11 37 50](https://user-images.githubusercontent.com/7756284/76613871-438e1d00-651f-11ea-9449-5853a2fab324.png)